Feature/xmlparser

ExoticWifi opened the pull request:
I want to keep the block of commented okhttp code to remember how to make it asynchronous again

Commit:
XMLParser

parsesxml

Commit:
renameditemsclass

renamed to articles

ExoticWifi committed:
fixed some features

okhttp requests are synchronous so that they will return the parsed xml to the mainActivity file. I want to keep the block of commented code so I can remember what the asynchronous okhttp code looks like

Nik-MS commented on app/build.gradle:
```diff
@@ -46,6 +46,6 @@ dependencies {
     implementation "org.jetbrains.kotlin:kotlin-script-runtime:1.3.72"
-
+    implementation 'org.jetbrains.kotlinx:kotlinx-coroutines-core:1.3.9'
 
```
Removed extra whitespace

Nik-MS commented on app/src/main/java/com/example/reflector_android/MainActivity.kt:
```diff
@@ -14,7 +16,11 @@ class MainActivity : AppCompatActivity() {
+            val service = RSSService().fetchNews()
+            println(service[2])
+        }
+
```
Remove extra whitespace.

Nik-MS commented on app/src/main/java/com/example/reflector_android/network/RSSService.kt:
```diff
@@ -1,22 +1,36 @@
-                val body = response.body?.string()
-                println(body)
-            }
+//        client.newCall(request).enqueue(object: Callback {
```
Remove unused code 14-26. If you need to recover this you can look back on your branch history.

Nik-MS commented on app/src/main/java/com/example/reflector_android/network/RSSService.kt:
```diff
@@ -1,22 +1,36 @@
+        }
+        return articles
     }
+
```
Remove extra whitespace.

Nik-MS commented on app/src/main/java/com/example/reflector_android/network/XmlParser.kt:
```diff
@@ -0,0 +1,135 @@
+import org.xmlpull.v1.XmlPullParserException
+import java.io.InputStream
+import java.lang.IllegalStateException
+
```
Remove one extra whitespace. Maintain a max of 1 line of space.

Nik-MS commented on app/src/main/java/com/example/reflector_android/network/XmlParser.kt:
```diff
@@ -0,0 +1,135 @@
+
+
+class XmlParser {
+    private val ns: String? = null
```
Rename ns to be more clear on what this is.

Nik-MS commented on app/src/main/java/com/example/reflector_android/network/XmlParser.kt:
```diff
@@ -0,0 +1,135 @@
+    suspend private fun readFeed(parser: XmlPullParser): MutableList<Article> {
+        val items = mutableListOf<Article>()
+
+        parser.require(XmlPullParser.START_TAG, ns, "rss")
```
Move "rss" to a Constant. Typically in programming, you want to minimize stringly-typed stuff because you're more likely to misspell "rss" if you need to move stuff around since it's not autocompleted.

Also if further down the road our API decides to change the XML tag for "item" to "object" we only have to change your constant declaration and that's it.

At the top of your XmlParser class, do this:
```
class XmlParser {
   companion object Tags {
        val rss = "rss"
        val item = "item"
        var title = "title"
        ....
    }
}
```

Then anywhere you type strings for "rss" or "item" or whatever you can just do Tags.[tag_you_want].

Nik-MS commented on app/src/main/java/com/example/reflector_android/network/XmlParser.kt:
```diff
@@ -0,0 +1,135 @@
+// to their respective "read" methods for processing. Otherwise, skips the tag.
+    @Throws(XmlPullParserException::class, java.io.IOException::class)
+    suspend fun readItem(parser: XmlPullParser): Article {
+        parser.require(XmlPullParser.START_TAG, ns, "item")
```
Move "item" to constant

Nik-MS commented on app/src/main/java/com/example/reflector_android/network/XmlParser.kt:
```diff
@@ -0,0 +1,135 @@
+                continue
+            }
+            when (parser.name) {
+                "title" -> title = readTitle(parser)
```
Move the strings to constants if possible.

Nik-MS commented on app/src/main/java/com/example/reflector_android/network/XmlParser.kt:
```diff
@@ -0,0 +1,135 @@
+    //processes title tag
+    @Throws(java.io.IOException::class, XmlPullParserException::class)
+    private fun readTitle(parser: XmlPullParser): String {
+        parser.require(XmlPullParser.START_TAG, ns, "title")
```
Move "title" to constant

Nik-MS commented on app/src/main/java/com/example/reflector_android/network/XmlParser.kt:
```diff
@@ -0,0 +1,135 @@
+    //processes link tags
+    @Throws(java.io.IOException::class, XmlPullParserException::class)
+    private fun readLink(parser: XmlPullParser): String {
+        parser.require(XmlPullParser.START_TAG, ns, "link")
```
Move "link" string to constant

Nik-MS commented on app/src/main/java/com/example/reflector_android/network/XmlParser.kt:
```diff
@@ -0,0 +1,135 @@
+
+    @Throws(java.io.IOException::class, XmlPullParserException::class)
+    private fun readPubDate(parser: XmlPullParser): String {
+        parser.require(XmlPullParser.START_TAG, ns, "pubDate")
```
Move "pubDate" to constant

Nik-MS commented on app/src/main/java/com/example/reflector_android/network/XmlParser.kt:
```diff
@@ -0,0 +1,135 @@
+    //processes description tag
+    @Throws(java.io.IOException::class, XmlPullParserException::class)
+    private fun readDescription(parser: XmlPullParser): String {
+        parser.require(XmlPullParser.START_TAG, ns, "description")
```
Move "description" to constant

Nik-MS commented on app/src/main/java/com/example/reflector_android/network/XmlParser.kt:
```diff
@@ -0,0 +1,135 @@
+        }
+    }
+}
+
```
Remove extra whitespace lines below

Nik-MS commented on app/src/main/java/com/example/reflector_android/network/XmlParser.kt:
```diff
@@ -0,0 +1,135 @@
+    }
+
+    // Parses the contents of an entry. If it encounters a title, summary, or link tag, hands them off
+// to their respective "read" methods for processing. Otherwise, skips the tag.
```
Indent this line

ExoticWifi committed:
Code Review Updates

I removed unnecessary whitespace and unused code. I also replaced the strings in the xml parser to constants

Nik-MS reviewed (approved):
Awesome Job man!
